.github/workflows/test-api.yaml

```diff
@@ -0,0 +1,90 @@
+name: API Tests
+on:
+  workflow_dispatch:
+    inputs:
+      focus:
+        description: 'Test suite to run'
+        required: false
+        default: 'All'
+        type: choice
+        options:
+        - 'All'
+        - 'Core Cluster Management'
+        - 'Discovery and Metadata'
+        - 'Security and Authentication'
+        - 'Machine Operations'
+      parallel:
```

Comment:
If we're making this an option, my guess is the default should be `true` (complete faster when run automatically), with the alternative `false` being if you want to trigger it manually and watch the logs scroll past.